### PR TITLE
Initial filter values

### PR DESCRIPTION
**Pull request by CampbellTimm**

### CODE REVIEW

File: examples/src/index.js [hunk 1]
```diff
@@ -32,6 +34,7 @@ const OptionGroups = {
 
 const App = () => (
     <FacetedSelect
+        initialValues={[{label: "First Name:Jane", value: "First Name:Jane"}]}
```
Comment (reecefenwick) [inline examples/src/index.js]: I think this should be initialValues={[{label: "First Name", value: "Jane"}]}

File: src/components/FacetedSelect.js [hunk 2]
```diff
@@ -35,7 +35,12 @@ class FacetedSelect extends React.Component {
...
-        onOptionsChanged: PropTypes.func.isRequired
+        onOptionsChanged: PropTypes.func.isRequired,
+        intialValues: PropTypes.arrayOf(PropTypes.object)
+    };
+
+    static defaultProps = {
+        initialValues: []
     };
 
     state = {
```
Comment (reecefenwick) [inline src/components/FacetedSelect.js:46]: I think you can init the state from props like this:

```javascript
state = {
  selectedvalues: this.props.initialValues || []
}
```
@cmonaghan1 thoughts?

File: examples/src/index.js (hunk 1 above)
Comment (CampbellTimm) [reply examples/src/index.js]: Im pretty sure it doesn't need a value now that I think about it, onOptionsChanged only uses the label.

File: src/__tests__/FacetedSelect.spec.js [hunk 3]
```diff
@@ -66,9 +75,16 @@ describe('FacetedSelect #handleChange', () => {
         wrapper = shallow(<FacetedSelect
             options={options}
             onOptionsChanged={onOptionsChangedMock}
+            initialValues={initialValues}
```
Comment (cmonaghan1) [inline src/__tests__/FacetedSelect.spec.js]: You'd want tests for when this isn't provided as well

File: src/components/FacetedSelect.js [hunk 4]
```diff
@@ -128,6 +136,10 @@ class FacetedSelect extends React.Component {
         )
     };
 
+    componentDidMount(){
+        this.onOptionsChanged(this.state.selectedValues);
```
Comment (reecefenwick) [inline src/components/FacetedSelect.js]: What is this for?

File: src/components/FacetedSelect.js [hunk 5]
```diff
@@ -35,12 +35,20 @@ class FacetedSelect extends React.Component {
...
         })).isRequired,
-        onOptionsChanged: PropTypes.func.isRequired
+        onOptionsChanged: PropTypes.func.isRequired,
+        initialValues: PropTypes.arrayOf(PropTypes.object)
     };
 
     state = {
         inputValue: '',
-        selectedValues: []
+        selectedValues: this.props.initialValues
```
Comment (reecefenwick) [inline src/components/FacetedSelect.js]: You could define static defaultProps for initialValues as an empty array to avoid a ternary here.

**Review by reecefenwick — APPROVED** [review]

**Pull request merged**